### PR TITLE
Fix #1205: Stop attaching logs to send a frown email

### LINKED ISSUE #1205: Stop attaching logs to send a frown email
change text in email to say:

```
Please briefly describe what you were doing that led to the issue if applicable. 



We have created a log in %TEMP%\RTVSLogs.zip. We have also opened an Explorer Window to this directory, with the file selected. Beware that these logs may contain sensitive information, such as your complete command history and variable values. 
```

### PR DESCRIPTION
**Pull request by int19h**
Always pop up Explorer with the log file selected, even when using Outlook.

Stop automatically attaching logs to email even when using Outlook, and change the text to give directions on how to do so manually.

Add basic diagnostic info (VS, RTVS and R versions) directly into the precomposed email body, so that it is present even if logs are not attached.

Use a custom non-modal window instead of the stock message box as a mailto: fallback, so that the entire body of the message is included there, and can be easily copy/pasted, in case mailto: handler didn't work for any reason.

### CODE REVIEW

File: src/Package/Impl/Feedback/SendMailFallbackWindow.xaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+<Window x:Class="Microsoft.VisualStudio.R.Package.Feedback.SendMailFallbackWindow"
+        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+        xmlns:local="clr-namespace:Microsoft.VisualStudio.R.Package.Feedback"
+        mc:Ignorable="d"
+        Title="Send RTVS Feedback" Height="400" Width="600" WindowStartupLocation="CenterOwner">
+    <DockPanel LastChildFill="True">
+        <TextBlock x:Uid="Header" DockPanel.Dock="Top" Margin="10,10,10,5" TextWrapping="Wrap">
```
Comment (int19h) [inline src/Package/Impl/Feedback/SendMailFallbackWindow.xaml]: @jflam, can you please take a look at the wording below and see if there's anything you'd like to change?

File: src/Package/Impl/Resources.resx [hunk 2]
```diff
@@ -582,8 +582,8 @@ Supported formats are: jpg/jpeg, png, tif/tiff, bmp.</value>
   <data name="Settings_TriggerOnTab_Description" xml:space="preserve">
     <value>If set to on completion list can be invoked by typing one or more characters and pressing Tab.</value>
   </data>
-  <data name="MailToFrownMessage" xml:space="preserve">
-    <value>If e-mail client does not open, please create an e-mail message to rtvsuserfeedback@microsoft.com and attach RTVSLogs.zip file that can be found in {0}. Briefly describe what you were doing that led to the issue if applicable.{1}Please be aware that the data contained in the attached logs contain your command history as well as all output displayed in the R Interactive Window.</value>
+  <data name="SendMailFallbackMessage" xml:space="preserve">
```
Comment (int19h) [inline src/Package/Impl/Resources.resx]: @jflam, also here:

File: src/Package/Impl/Resources.resx [hunk 3]
```diff
@@ -606,4 +606,17 @@ Supported formats are: jpg/jpeg, png, tif/tiff, bmp.</value>
   <data name="Warning_UncPath" xml:space="preserve">
     <value>The project appears to be on a network share. R Tools for Visual Studio may not work correctly when files are on a remote share. </value>
   </data>
+  <data name="SendFrownEmailBody" xml:space="preserve">
```
Comment (int19h) [inline src/Package/Impl/Resources.resx]: @jflam, and here:

**Comment by int19h** [comment]
Here's what the new fallback window (in case `mailto:` is used and doesn't work) looks like. This also shows the verbiage:

image

**Comment by int19h** [comment]
(If Outlook is detected and used, you don't get the window - it just opens a new email with the same exact text as the body.)

File: src/Package/Impl/Feedback/SendFrownCommand.cs [hunk 4]
```diff
@@ -23,7 +18,11 @@ internal sealed class SendFrownCommand : SendMailCommand {
 
         protected override void Handle() {
             string zipPath = DiagnosticLogs.Collect();
-            SendMail(_disclaimer, "RTVS Frown", zipPath);
+
+            var generalData = new StringWriter();
+            DiagnosticLogs.WriteGeneralData(generalData, detailed: false);
+
+            SendMail(string.Format(Resources.SendFrownEmailBody, zipPath, generalData.ToString(), zipPath), "RTVS Frown", zipPath);
```
Comment (huguesv) [inline src/Package/Impl/Feedback/SendFrownCommand.cs]: SendFrownEmailBody only has 2 replacement tags, you are passing in 3 parameters (zip path twice).

File: src/Package/Impl/Feedback/SendMailCommand.cs [hunk 5]
```diff
@@ -27,48 +40,29 @@ internal class SendMailCommand : PackageCommand {
...
-                    VsAppShell.Current.ShowMessage(body, MessageButtons.OK);
-                }
+                VsAppShell.Current.DispatchOnUIThread(() => {
+                    var fallbackWindow = new SendMailFallbackWindow {
+                        MessageBody = body
+                    };
+                    fallbackWindow.Show();
+                });
+
+                //VsAppShell.Current.ShowMessage(Resources.SendMailFallbackMessage + "\r\n\r\n" + body, MessageButtons.OK);
```
Comment (huguesv) [inline src/Package/Impl/Feedback/SendMailCommand.cs]: Commented out code can be removed?

Comment (int19h) [inline src/Package/Impl/Feedback/SendMailCommand.cs]: Yep.

**Comment by jflam** [comment]
Looks good! I think a Cancel button on the dialog would also be appropriate?

**Comment by int19h** [comment]
The dialog is non-modal, so there isn't really anything to cancel - it's more of a notification. I can add "Close" though.

**Comment by int19h** [comment]
Addressed feedback.

**Comment by huguesv** [comment]
:+1:

**Pull request merged**